### PR TITLE
Delete Associated Service

### PR DESCRIPTION
**Pull request by kiran-mastek**
Delete an Associated Service

Story: [AB#13376](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13376)
Task: [AB#14108](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/14108), [AB#14109](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/14109)

### CODE REVIEW

File: app/pages/sections/order-items/associated-services/delete/ui-tests/general.ui.test.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  await t
+    .expect(await extractInnerText(title)).eql('Delete write-on-time-associated-service from order-id?');
+});
+
+test('should render the description', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const description = Selector('h2[data-test-id="delete-catalogue-page-description"]');
+
```
Comment (NickNewlands) [inline app/pages/sections/order-items/associated-services/delete/ui-tests/general.ui.test.js:90]: I can't see the template in this commit but the description should be a `p` not a `h2` with the classes `nhsuk-heading-l nhsuk-u-font-size-22` to make it look the same.

File: app/pages/sections/order-items/associated-services/delete/routes.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    logger.info(`navigating to catalogue ${solutionName} delete-associated-service-confirmation page`);
+    res.render('pages/sections/order-items/catalogue-solutions/delete/confirmation/template.njk', addContext({ context, user: req.user, csrfToken: req.csrfToken() }));
+  }));
+
+  router.post('/:catalogueItemId/confirmation/:solutionName/continue', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
+    const { orderId } = req.params;
+    return res.redirect(`${config.baseUrl}/organisation/${orderId}/associated-services`);
+  }));
+
```
Comment (NickNewlands) [inline app/pages/sections/order-items/associated-services/delete/routes.js:60]: Could this be a link? It does not look like you are handling any post data, the `orderId` must be somewhere on the page to be posted.

File: app/pages/sections/order-items/associated-services/delete/ui-tests/general.ui.test.js (hunk 1 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/associated-services/delete/ui-tests/general.ui.test.js:90]: This template is used in the form for all types - catalogue solutions, additional/associated services so - the change will require changing tests for all the calls and should be a new task/story.

Comment (NickNewlands) [reply app/pages/sections/order-items/associated-services/delete/ui-tests/general.ui.test.js:90]: Ok, I will add a task for it on technical improvements, it is part of the accessibility audit.

File: app/pages/sections/order-items/associated-services/delete/routes.js (hunk 2 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/associated-services/delete/routes.js:60]: Possibly could do with a link but for consistency should be done for all types mentioned above in a separate new ticket.

**Review by NickNewlands — APPROVED** [review]

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**